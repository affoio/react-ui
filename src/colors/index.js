const colors = {
  indigoBlue: {
    800: '#4718b4',
    700: '#623abf',
    600: '#8769ce',
    500: '#ac97dd',
    400: '#d1c5ec',
    200: '#ece7f7',
    300: '#e4ddf4',
    100: '#f6f3fb',
  },
  marigold: {
    800: '#ffca00',
    700: '#ffd226',
    600: '#ffdc59',
    500: '#ffe78c',
    400: '#fff2bf',
    300: '#fff7d9',
    200: '#fff9e5',
    100: '#fffcf2',
  },
  cerise: {
    800: '#e2086e',
    700: '#e62d84',
    600: '#ec5ea1',
    500: '#f290be',
    400: '#f8c1db',
    300: '#fbdaea',
    200: '#fce6f0',
    100: '#fdf2f8',
  },
  black: {
    800: '#000000',
    700: '#262626',
    600: '#595959',
    500: '#8c8c8c',
    400: '#bfbfbf',
    300: '#d9d9d9',
    200: '#e5e5e5',
    100: '#f2f2f2',
  },
  seaweed: {
    800: '#13c64f',
    700: '#36ce69',
    600: '#65da8c',
    500: '#95e5b0',
    400: '#c4f1d3',
    300: '#dcf7e5',
    200: '#e7f9ed',
    100: '#f3fcf6',
  },
  white: '#ffffff',
};

export default colors;
