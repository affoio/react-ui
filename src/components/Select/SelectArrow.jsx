import React from 'react';
import { bool } from 'prop-types';
import styled from 'styled-components';

import { addStyleIfPropTruly } from '../../helpers/styledComponents';

import ArrowUpIcon from '../Icon/ArrowUpIcon';
import SelectAddon from './SelectAddon';
import colors from '../../colors';

export const SelectArrowStyled = styled(ArrowUpIcon)`
  transform: rotate(0);
  transition: transform 0.2s linear;
  will-change: transform;
  ${addStyleIfPropTruly('isOpen', 'transform: rotate(-180deg)')}
`;

const propTypes = {
  isOpen: bool,
  active: bool,
};

const SelectArrow = ({ isOpen, active }) => {
  const color = active ? colors.black[800] : colors.black[400];

  return (
    <SelectAddon position="right">
      <SelectArrowStyled {...{ color, isOpen }} />
    </SelectAddon>
  );
};

SelectArrow.propTypes = propTypes;

export default SelectArrow;
