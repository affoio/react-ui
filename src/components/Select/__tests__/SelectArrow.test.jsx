import React from 'react';
import renderer from 'react-test-renderer';
import SelectArrow from '../SelectArrow';
import colors from '../../../colors';

describe('<SelectArrow', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<SelectArrow />);

    expect(tree).toMatchSnapshot();
  });

  it('rotates when isOpen', () => {
    const tree = renderer.create(<SelectArrow isOpen />).toJSON();

    expect(tree).toHaveStyleRule('transform: rotate(-180deg)');
  });

  it('changes color when active', () => {
    const tree = renderer.create(<SelectArrow active />).toJSON();

    expect(tree).toHaveStyleRule(`color: ${colors.black[800]}`);
  });
});
