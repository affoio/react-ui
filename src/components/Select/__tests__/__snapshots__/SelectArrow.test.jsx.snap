// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectArrow renders correctly 1`] = `
.c1 {
  fill: #bfbfbf;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-left: 4px;
  width: 24px;
  height: 24px;
}

.c2 {
  -webkit-transform: rotate(0);
  -ms-transform: rotate(0);
  transform: rotate(0);
  -webkit-transition: -webkit-transform 0.2s linear;
  -webkit-transition: transform 0.2s linear;
  transition: transform 0.2s linear;
  will-change: transform;
}

<div
  className="c0"
>
  <svg
    className="c1 c2"
    color="#bfbfbf"
    height="24"
    viewBox="0 0 24 24"
    width="24"
  >
    <path
      d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
    />
  </svg>
</div>
`;
